core/src/Kokkos_CudaSpace.hpp

```diff
@@ -610,6 +630,10 @@ class SharedAllocationRecord<Kokkos::CudaSpace, void>
       const size_t arg_alloc_size,
       const RecordBase::function_type arg_dealloc = &base_t::deallocate);
 
+  // helper function to work around MSVC+NVCC issue
+  // https://github.com/kokkos/kokkos/issues/5258
+  void deep_copy_header_no_exec(void*, void*);
```

Comment:
```suggestion
  void deep_copy_header_no_exec(void*, void const*);
```

Reply:
I was about to comment there is no reason to add here once you move the definitions to the source file but since we won't do it in this PR, please declare that guy as a static member function.